File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.cpp
class Solution {
public:
    int minimumCardPickup(vector<int>& c) {
        for(int i=0;i<c.size()-1;i++){
            if(c[i]==c[i+1]) return 2;
        }
        int ans=INT_MAX;
        unordered_map<int,int>mp;
        for(int i=0;i<c.size();i++){
            if(mp.count(c[i])>0) //always remember
                ans=min(ans,abs(i-mp[c[i]])+1);
            mp[c[i]]=i;
        }
        return ans==INT_MAX ? -1 : ans;
    }
};

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        dp = dict()
        result = 10**5 + 1
        for i in range(len(cards)):
            if cards[i] in dp:
                result = min(result, i - dp[cards[i]] + 1)
            dp[cards[i]] = i
        return -1 if result == 10**5 + 1 else result